docs/pages/CodeAnalysis.md

```diff
@@ -0,0 +1,226 @@
+
+### XML Composite Analyzer
+
+The **XMLComposite** control is deprecated since UI5 Version 1.88. Nevertheless UI5 Tooling will attempt to analyze the declaration of any such controls in a project. The [XML Composite Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/XMLCompositeAnalyzer.js) searches for the name of the configured fragment containing the **XMLComposite** control.
+
+=== "Name of the XMLComposite is equal with fragment name"
+
+    ```javascript hl_lines="4"
+    sap.ui.define([
+        "sap/ui/core/XMLComposite"
+    ], function(XMLComposite) {
+        return XMLComposite.extend("composites.MyComposite", {} 
+    });
+    ```
+
+=== "Dedicated fragment name"
+
+    ```javascript hl_lines="5"
+    sap.ui.define([
+        "sap/ui/core/XMLComposite"
+    ], function(XMLComposite) {
+        return XMLComposite.extend("composites.MyComposite", {
+            fragment: "composites.custom.MyComposite"
+        } 
+    });
+    ```
+
+## Library Initialization
+
+The [library.js Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/analyzeLibraryJS.js) checks every `library.js` file in the namespace of a library for occurences of a `sap/ui/core/Core#initLibrary` call. If so, the following information will be placed in the generated manifest.json:
+
+- noLibraryCSS: false when the noLibraryCSS property had been set in the initLibrary info object
+- types: string array with the names of the types contained in the library
+- controls: string array with the names of the controls defined in the library
+- elements: string array with the names of the elements defined in the library
+- interfaces: string array with the names of the interfaces defined in the library
+
+When using `sap/ui/core/Core#initLibrary` requires the usage of "Simple Literals" for the parameter passed to this function call.
+
+Note: currently only the usage via the global `sap.ui.getCore().initLibrary` is supported by this analyzer. Requiring the `sap/ui/core/Core` and then call `Core.initLibrary` is not recognized by this analyzer.
+
+## JSDoc
+
+The UI5 Tooling offers a JSDoc build, which is enhanced by UI5 specific JSDoc features.
+
+An UI5 SDK can be build locally. To get more insights about the local UI5 SDK build setup, have a look at the [Developer Guide](https://github.com/SAP/openui5/blob/master/docs/developing.md#building-the-openui5-sdk-demo-kit).
+
+Currently, the resources needed for a UI5 SDK build are stored in [openui5](https://github.com/SAP/openui5/tree/master/lib/jsdoc) and in [ui5-builder](https://github.com/SAP/ui5-builder/blob/main/lib/processors/jsdoc). This double maintenance is needed because these files are not part of the `sap.ui.core` library artefact, so building JSDoc for any library has no access to the needed resources. Therefore it is necessary to have these resources also available in the *ui5-builder*. This might change in future.
```

Comment:
```suggestion
Currently, the resources needed for a UI5 SDK build are stored in [openui5](https://github.com/SAP/openui5/tree/master/lib/jsdoc) and in [ui5-builder](https://github.com/SAP/ui5-builder/blob/main/lib/processors/jsdoc). This double maintenance is needed because these files are not part of the `sap.ui.core` library artefact, so building JSDoc for any library has no access to the needed resources. It's therefore necessary to have these resources also available in the *ui5-builder*. This might change in the future.
```